README.md

```diff
@@ -8,6 +8,17 @@ This extension is configured to talk to a language server over WebSockets. If yo
 private Sourcegraph instance, you should run your own language server. The server is available as a
 Docker image `sourcegraph/lang-typescript` from Docker Hub.
 
+### 🚨 Secure deployment 🚨
+
+We recommend deploying the language server behind an auth proxy or firewall and
+treating it like an authenticated user because anyone that connects to the
+language server can access resources such as private code that the language
+server has access to.
```

Comment:
I find this sentence a bit confusing. The language server doesn’t have access to anything by default, and what does “treating it as an authenticated user” mean?

Reply:
My wording definitely wasn't clear. My intent was to recommend against deploying the language server inside a Kubernetes cluster where the rest of Sourcegraph is running (or anywhere that sensitive data is accessible) because it seems to me (and correct me if I'm wrong) that language server can be forced to make an HTTP GET to an arbitrary URL by any user who is connected to it by setting the `rootUri`.

https://github.com/sourcegraph/sourcegraph-typescript/blob/cfd79574028ebd50842cfeb0ba5685a18d6bb02f/server/src/server.ts#L612

This applies to the Go and Python language servers, too.

I opened a Slack discussion about this https://sourcegraph.slack.com/archives/C07KZF47K/p1551298666152700 to determine next steps.